### PR TITLE
feat: gzip

### PR DESCRIPTION
**Pull request by poolpitako**

### CODE REVIEW

**Comment by bh2smith** [comment]
Hey @poolpitako - thanks for the contribution. Would you mind explaining a bit the motivation for this and what it is actually for?

**Comment by poolpitako** [comment]
@bh2smith enabling gzip support for all endpoints. Default is 500 bytes.
Example in the solver I am working on:

curl without gzip headers:
```
curl -vX POST "http://127.0.0.1:8000/solve" -H  "accept: application/json" -H  "Content-Type: application/json" --data "@/Users/user/dev/yswap-solver/data/one_inch_example.json"
```

response:
```
< content-length: 25203
< content-type: application/json
```

curl with gzip headers:
```
curl -vX POST "http://127.0.0.1:8000/solve" -H "Accept-Encoding: gzip" -H  "accept: application/json" -H  "Content-Type: application/json" --data "@/Users/user/dev/yswap-solver/data/one_inch_example.json"
```

response:
```
< content-length: 1597
< content-type: application/json
< content-encoding: gzip
```

**Comment by bh2smith** [comment]
Looks good then. Does you second example need to be a zip file rather than json?

**Review by bh2smith — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ venv/
 
 .idea/
 .vscode/
+.DS_Store
```
Comment (bh2smith) [inline .gitignore:8]: I'll allow it, just because we're already have IDE specifics in the ignore file.

**Pull request merged**